src/JSTest.Runner/Environment/Node/DebugLogger.ts

```diff
@@ -16,42 +20,48 @@ export class DebugLogger implements IDebugLogger {
     }
 
     private defaultLogFileName() {
-        const date = new Date();
-        const dateStamp = `${date.getFullYear()}-${date.getMonth()}-${date.getDate()}`;
-        const timeStamp = `${date.getHours()}-${date.getMinutes()}.${date.getMilliseconds()}`;
+        const [dateStamp, timeStamp] = GetTimeStamp('-', '.');
         return `jstest.${this.processName}.${dateStamp}-${timeStamp}_${process.pid}`;
     }
 
     private defaultPath() {
-
         return path.join(process.cwd(), this.defaultLogFileName() + this.defaultLogFileExtension());
     }
 
     public initialize(diagFile: string) {
         let diagFilePath = '';
         
-        if (!diagFile) {
-            diagFilePath = this.defaultPath();
-        }
-
-        try {
-            const stat = fs.lstatSync(diagFile);
-            if (stat.isDirectory()) {
-                diagFilePath = path.join(diagFile, this.defaultLogFileName() + this.defaultLogFileExtension());
-            }
-        } catch (e) {
+        if (diagFile) {
             try {
-                const dirName = path.dirname(diagFile);
-                const stat = fs.lstatSync(dirName);
+                const stat = fs.lstatSync(diagFile);
                 if (stat.isDirectory()) {
-                    diagFilePath = path.join(dirName, this.defaultLogFileName() + '-' + path.basename(diagFile));
+                    diagFilePath = path.join(diagFile, this.defaultLogFileName() + this.defaultLogFileExtension());
                 }
             } catch (e) {
-                diagFilePath = this.defaultPath();
+                try {
+                    const dirName = path.dirname(diagFile);
+                    const stat = fs.lstatSync(dirName);
+                    if (stat.isDirectory()) {
+                        diagFilePath = path.join(dirName, this.defaultLogFileName() + '-' + path.basename(diagFile));
+                    }

```

Comment:
can we avoid this code.. and ensure we create a new file each time..?

Reply:
`this.logFileStream = fs.createWriteStream(diagFilePath);` ensures that, this logic is to check whether the option contained a directory or a file, if it is a file say of file name D:\a\b.log, the file that's created is D:\a\<stamp>-b.log